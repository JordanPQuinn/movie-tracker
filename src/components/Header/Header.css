h1 {
  font-family: 'Anton', sans-serif;
  padding-left: 10px;
  border-top-left-radius: 19px;
  border-bottom-left-radius: 19px;

  background: -moz-linear-gradient(left, rgba(255,255,255,1) 0%, rgba(255,255,255,0) 100%);
  background: -webkit-linear-gradient(left, rgba(255,255,255,1) 0%,rgba(255,255,255,0) 100%);
  background: linear-gradient(to right, rgba(255,255,255,1) 0%,rgba(255,255,255,0) 100%);
}

.logo {
  display: inline-block;
  padding: 0 0 17px 0;
  width: 50px;
}

.title {
  color: #FFFFFF;
}

.title-cont {
  display: flex;
  flex-direction: row;
  align-items: center;
  align-content: center;
  padding: 15px;
  background-color: #262626;
}

#menu {
  position: absolute;
  top: 10px;
  right: 10px;
  color: white;
  font-size: 25px;
}