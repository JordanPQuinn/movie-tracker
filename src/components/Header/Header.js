import React from 'react';
import camera from '../../images/video-camera.svg';
import { Link } from 'react-router-dom';
import './Header.css';

      // <i className="fas fa-bars" id="menu" />


export const Header = () => {
  return (
    <header>
      <Link to='/'>
      <div className="title-cont">
        <img className="logo" alt="logo" src={camera} />
        <h1>Movie<span className="title">Tracker</span></h1>
      </div>
      </Link>
      <Link to='/login'>Login</Link>
      <br />
      <Link to='/sign-up'>Create Account</Link>
    </header>
  );
};
