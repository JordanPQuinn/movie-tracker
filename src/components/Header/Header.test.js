import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';
import { mapStateToProps, mapDispatchToProps } from './Header';

//can we test links?

describe('Header', () => {
  let renderedHeader;
  let mockSignOut

  beforeEach(() => {
    const user = {name:'person'}
    mockSignOut = jest.fn()
    renderedHeader = shallow(<Header user={user} signOut={mockSignOut}/>);
  });

  it('should match the snapshot', () => {
    expect(renderedHeader).toMatchSnapshot();
  });

  it('should MSTP, taking in the store and mapping our user to that container', () => {
    const mockStore = {
      user: {
        name: 'Matt', 
        email: '123@gmail.com', 
        password: 'password'
      }
    }
    const mappedStore = mapStateToProps(mockStore)
    expect(mappedStore.user).toEqual(mockStore.user)
  })

 it('should call the dispach fn when using a fn from mdtp', () => {
  const mockDispatch = jest.fn()
  const mapped = mapDispatchToProps(mockDispatch)

  mapped.signOut()
  expect(mockDispatch).toHaveBeenCalled()
  });

  it('should call signOut on click of our button', () => {
    renderedHeader.find('#sign-out').simulate('click');

    expect(mockSignOut).toHaveBeenCalled()
  });
});
