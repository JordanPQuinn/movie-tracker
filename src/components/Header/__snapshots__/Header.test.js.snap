// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    signOut={[Function]}
    user={
        Object {
            "name": "person",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="title-cont"
          replace={false}
          to="/"
>
          <img
                    alt="logo"
                    className="logo"
                    src="video-camera.svg"
          />
          <h1>
                    Movie
                    <span
                              className="title"
                    >
                              Tracker
                    </span>
          </h1>
</Link>,
        <div>
          <h2>
                    Hello, 
                    person
                     
          </h2>
          <button
                    onClick={[Function]}
                    type="submit"
          >
                     Sign Out 
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="logo"
              src="video-camera.svg"
/>,
            <h1>
              Movie
              <span
                            className="title"
              >
                            Tracker
              </span>
</h1>,
          ],
          "className": "title-cont",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "logo",
              "src": "video-camera.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Movie",
                <span
                  className="title"
>
                  Tracker
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Movie",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tracker",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Tracker",
                "type": "span",
              },
            ],
            "type": "h1",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Hello, 
              person
               
</h2>,
            <button
              onClick={[Function]}
              type="submit"
>
               Sign Out 
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hello, ",
                "person",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Hello, ",
              "person",
              " ",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Sign Out ",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": " Sign Out ",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="title-cont"
            replace={false}
            to="/"
>
            <img
                        alt="logo"
                        className="logo"
                        src="video-camera.svg"
            />
            <h1>
                        Movie
                        <span
                                    className="title"
                        >
                                    Tracker
                        </span>
            </h1>
</Link>,
          <div>
            <h2>
                        Hello, 
                        person
                         
            </h2>
            <button
                        onClick={[Function]}
                        type="submit"
            >
                         Sign Out 
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="logo"
                src="video-camera.svg"
/>,
              <h1>
                Movie
                <span
                                className="title"
                >
                                Tracker
                </span>
</h1>,
            ],
            "className": "title-cont",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "logo",
                "src": "video-camera.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Movie",
                  <span
                    className="title"
>
                    Tracker
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Movie",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Tracker",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": "Tracker",
                  "type": "span",
                },
              ],
              "type": "h1",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Hello, 
                person
                 
</h2>,
              <button
                onClick={[Function]}
                type="submit"
>
                 Sign Out 
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hello, ",
                  "person",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Hello, ",
                "person",
                " ",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Sign Out ",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": " Sign Out ",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
